File: loja/pedidos/src/main/java/br/com/bota/repository/PedidoRepository.java
package br.com.bota.repository;

import br.com.bota.entity.Pedido;
import br.com.bota.entity.enums.PedidoStatusEnum;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Set;
import java.util.UUID;

@Repository
public interface PedidoRepository extends JpaRepository<Pedido, UUID> {
    List<Pedido> findByClienteIdOrderById(UUID id);

    List<Pedido> findByStatusIn(Set<PedidoStatusEnum> status);
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/repository/PedidoRepository.kt
package br.com.bota.repository

import br.com.bota.entity.Pedido
import br.com.bota.lojalib.enums.PedidoStatusEnum
import io.micronaut.data.annotation.Repository
import io.micronaut.data.jpa.repository.JpaRepository
import java.util.*

@Repository
interface PedidoRepository : JpaRepository<Pedido, UUID> {
    fun findByClienteIdOrderById(id: UUID): List<Pedido>
    fun findByStatusIn(status: Set<PedidoStatusEnum>): List<Pedido>
}